src/main/scala/is/hail/annotations/ReadRowsRDD.scala

```diff
@@ -1,21 +1,49 @@
 package is.hail.annotations
 
-import java.io.{DataInputStream, DataOutputStream}
+import java.io.{InputStream, OutputStream}
+import java.util
 
 import is.hail.expr._
 import is.hail.utils.{SerializableHadoopConfiguration, _}
-import net.jpountz.lz4.{LZ4BlockInputStream, LZ4Factory}
+import is.hail.variant.LZ4Utils
 import org.apache.commons.lang3.StringUtils
 import org.apache.spark.rdd.RDD
-import org.apache.spark.sql.Row
 import org.apache.spark.{Partition, SparkContext, TaskContext}
 
-class Decoder(in: DataInputStream) {
+final class Decoder() {
```

Comment:
somewhat of an aside: should `readRegionValue` be called `readCompoundData`? Isn't a region value just a memory and an offset? `readRegionValue` seems to be a method for reading non-primitive types.